src/hotspot/share/opto/vectornode.hpp

```diff
@@ -1681,21 +1681,26 @@ class VectorBoxNode : public InlineTypeNode {
   static VectorBoxNode* make_box_node(PhaseGVN& gvn, Compile* C, Node* box, Node* val,
                                       const TypeInstPtr* box_type, const TypeVect* vt) {
     ciInlineKlass* vk = static_cast<ciInlineKlass*>(box_type->inline_klass());
-    VectorBoxNode* box_node = new VectorBoxNode(vk, box, true, vk->is_empty() && vk->is_initialized());
+    ciInlineKlass* payload = vk->declared_nonstatic_field_at(0)->type()->as_inline_klass();
+    Node* payload_oop = payload->is_initialized() ? default_oop(gvn, payload) : gvn.zerocon(T_PRIMITIVE_OBJECT);
+    Node* payload_value = InlineTypeNode::make_uninitialized(gvn, payload, true);
+    payload_value->as_InlineType()->set_field_value(0, val);
+    payload_value = gvn.transform(payload_value);
+    VectorBoxNode* box_node = new VectorBoxNode(vk, box, false, vk->is_empty() && vk->is_initialized());
```

Comment:
I wanted to get away with VectorBoxAllocateNode (a macro node) in a sub-sequent patch, I kept it currently because it being a SafePoint node carries JVMState for re-initialization of GraphKit during expansion, this helps in accessing  control and memory edges needed during new_instance creation for vector_boxes during expansion.

Otherwise, InlineTypeNode handle control flow merges inherently.
https://github.com/openjdk/valhalla/blob/lworld%2Bvector/src/hotspot/share/opto/inlinetypenode.hpp#L91